packages/amd-loader/amd-loader.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+(function() {
+
+function Registry() {
+  this.modules = {};
+  this.topLevelIdx = 0;
+  this.previousTopLevelUrl = undefined;
+  this.pendingDefine = undefined;
+}
+
+Registry.prototype.get = function(url) {
+  var module = this.modules[url];
+  if (module === undefined) {
+    module = this.modules[url] = new Module(url);
+  }
+  return module;
+};
+
+var registry = new Registry();
+
+function getBasePath(url) {
+  // TODO(aomarks) Maybe a better way to do this.
+  var lastSlash = url.lastIndexOf('/');
+  if (lastSlash === url.indexOf('://') + 2) {
+    // http://example.com -> http://example.com/
+    return url + '/';
+  }
+  // http://example.com/foo/bar.html -> http://example.com/foo/
+  return url.substring(0, lastSlash + 1);
```

Comment:
Also, if you have `URL`: `const getBasePath = url => new URL("./", url).href;`

Reply:
Worth noting that IE 11 does not have URL, but the webcomponents polyfill includes it.